Deprecated debugging enabler used

Started to see these lines while building Gallery app:

```
/Users/tomasmizera/Projects/mobile/input/gallery/main.cpp:49:24: warning: 'QQmlDebuggingEnabler' is deprecated: Use QQmlTriviallyDestructibleDebuggingEnabler instead or just call QQmlDebuggingEnabler::enableDebugging(). [-Wdeprecated-declarations]
  QQmlDebuggingEnabler enabler;
                       ^
/opt/Qt/6.6.0/macos/lib/QtQml.framework/Headers/qqmldebug.h:26:5: note: 'QQmlDebuggingEnabler' has been explicitly marked deprecated here
    QT_DEPRECATED_VERSION_X_6_4("Use QQmlTriviallyDestructibleDebuggingEnabler instead "
    ^
/opt/Qt/6.6.0/macos/lib/QtCore.framework/Headers/qtdeprecationmarkers.h:159:44: note: expanded from macro 'QT_DEPRECATED_VERSION_X_6_4'
# define QT_DEPRECATED_VERSION_X_6_4(text) QT_DEPRECATED_X(text)
                                           ^
/opt/Qt/6.6.0/macos/lib/QtCore.framework/Headers/qtdeprecationmarkers.h:27:33: note: expanded from macro 'QT_DEPRECATED_X'
#  define QT_DEPRECATED_X(text) Q_DECL_DEPRECATED_X(text)
                                ^
/opt/Qt/6.6.0/macos/lib/QtCore.framework/Headers/qcompilerdetection.h:982:36: note: expanded from macro 'Q_DECL_DEPRECATED_X'
#  define Q_DECL_DEPRECATED_X(x) [[deprecated(x)]]
                                   ^
1 warning generated.
```

--- a/app/CMakeLists.txt
+++ b/app/CMakeLists.txt
@@ -537,6 +537,10 @@ endif ()
 # QML
 # ########################################################################################
 
+target_compile_definitions(
+  Input PRIVATE $<$<OR:$<CONFIG:Debug>,$<CONFIG:RelWithDebInfo>>:QT_QML_DEBUG>
+)
+
 add_subdirectory(qml)
 
 # register Singleton TODO REMOVE!

--- a/app/main.cpp
+++ b/app/main.cpp
@@ -134,10 +134,6 @@
 
 #include <QQuickStyle>
 
-#ifdef QT_DEBUG
-#include <QQmlDebuggingEnabler>
-#endif
-
 #ifdef MOBILE_OS
 #include <QFile>
 #include <QDir>
@@ -391,9 +387,6 @@ int main( int argc, char *argv[] )
   QCoreApplication::setApplicationName( "Input" ); // used by QSettings
   QCoreApplication::setApplicationVersion( version );
 
-#ifdef QT_DEBUG
-  QQmlDebuggingEnabler enabler;
-#endif
 
 #ifdef INPUT_TEST
   InputTests tests;

--- a/gallery/CMakeLists.txt
+++ b/gallery/CMakeLists.txt
@@ -97,7 +97,9 @@ set_target_properties(
 target_link_libraries(MerginMapsGallery PRIVATE Qt6::Quick Qt6::Multimedia)
 target_include_directories(MerginMapsGallery PRIVATE ${CMAKE_SOURCE_DIR} ../app ../core)
 target_compile_definitions(
-  MerginMapsGallery PRIVATE GALLERY_SOURCE_DIR="${CMAKE_SOURCE_DIR}"
+  MerginMapsGallery
+  PRIVATE GALLERY_SOURCE_DIR="${CMAKE_SOURCE_DIR}"
+  PRIVATE $<$<OR:$<CONFIG:Debug>,$<CONFIG:RelWithDebInfo>>:QT_QML_DEBUG>
 )
 
 install(

--- a/gallery/main.cpp
+++ b/gallery/main.cpp
@@ -10,9 +10,6 @@
 #include <QGuiApplication>
 #include <QQmlApplicationEngine>
 #include <QQmlContext>
-#ifdef QT_DEBUG
-#include <QQmlDebuggingEnabler>
-#endif
 #ifdef DESKTOP_OS
 #include "hotreload.h"
 #endif
@@ -44,10 +41,6 @@ int main( int argc, char *argv[] )
   InputUtils iu;
 
   QQmlApplicationEngine engine;
-
-#ifdef QT_DEBUG
-  QQmlDebuggingEnabler enabler;
-#endif
 
   // Register C++ enums
   qmlRegisterUncreatableType<RegistrationError>( "mm", 1, 0, "RegistrationError", "RegistrationError Enum" );

